components/common/component.js

```diff
@@ -0,0 +1,17 @@
+export class Component {
+  constructor(children = []) {
+    this.children = children;
+  }
+
+  render() {
+    return "";
+  }
+
+  attachEvent() {
+    this.children.forEach((child) => {
+      if (child.attachEvent) {
+        child.attachEvent();
+      }
```

Comment:
P4;

옵셔널 체이닝을 사용하면 조금 더 코드를 쉽게 풀어낼 수 있어요!

```suggestion
      child.attachEvent?.();
```